model-test/src/main/java/org/jboss/as/model/test/FailedOperationTransformationConfig.java

```diff
@@ -437,10 +437,9 @@ public ModelNode correctOperation(ModelNode operation) {
                     } else {
                         op.get(attr).set(complexChildConfig.correctOperation(operation.get(attr)));
                     }
-                    return op;
                 }
             }
-            return operation;
+            return op;
```

Comment:
It might be safer to track whether 'op' has been modified (ie the 'if (checkValue...' block was entered) and return 'operation' if not.

The op.get call at L429 may modify 'op' without really changing its semantic content and then there is other code that checks if the return val from this method equals what was passed in. Having a different result from that check when there was no actual change might mess something up.

Reply:
fixed